src/main.c

```diff
@@ -127,16 +127,15 @@ main (int argc, char *argv[])
 
   }
   while (next_option != -1);
-
-  if (getenv ("HOME") == NULL)
+  
+  char *HOMEDIR = getenv ("HOME");
+  
+  if (HOMEDIR == NULL)
```

Comment:
Looks good, but let's "reverse" the condition. If HOMEDIR != NULL do the bufchk else return 1.